GitUI/CommandsDialogs/SettingsDialog/Pages/FormBrowseRepoSettingsPage.cs

```diff
@@ -32,5 +34,19 @@ public static SettingsPageReference GetPageReference()
         {
             return new SettingsPageReferenceByType(typeof(FormBrowseRepoSettingsPage));
         }
+
+        private void cboTerminal_SelectionChangeCommitted(object sender, System.EventArgs e)
+        {
+            if (!ShellHelper.ShellIsOnPath((string)cboTerminal.SelectedItem))
```

Comment:
I'm not sure what you mean here.

Reply:
Turn
```
void foo()
{
    if (cond1)
    {
        // do 1
        if (cond2)
        {
             // do 2
        }
    }
}
```
into
```
void foo()
{
    if (!cond1)
    {
         return;
    }

    // do 1

    if (!cond2)
    {
        return;
    }

    // do 2
}
```

Reply:
There's only one condition here -- if the shell isn't on the path, then reject the selection. IMO, this:
```csharp
void Foo()
{
  if (!okay)
  {
    reject;
  }
}
```
is clearer than:
```csharp
void Foo()
{
  if (okay)
  {
    return;
  }

  reject;
}
```

Reply:
It is our coding style and recommended by Resharper. (I am not completely happy with it but it is consistent and reduces complexity a little.)
In my example you see the benefit, which also exists in the simple case when the "do"/`reject` code is to be extended.

Reply:
Personally I disagree, but I'll make the change.

Reply:
I don't see the examples comparable to the code in question.
If the code was like in your examples, it would've been fine. However there was more than one instructions, i.e.
```cs
void Foo()
{
  if (!okay)
  {
    DoSomething();
    Reject;
  }
}
```
which is now:
```cs
void Foo()
{
  if (okay)
  {
    return;
  }

  DoSomething();
  Reject;
}
```

It is a styling preference, sure. However 'fast returns' make a big difference to readability and ability to reason about the code. It helps to reduce code rot as well - quite often one condition today turns into several nested conditions tomorrow. By reducing nesting it becomes easier to avoid nesting in the future.
And last but not least - fast returns also help in writing tests.

Reply:
I disagree about the readability of the two approaches for something like this, but it's your project and I'm happy to follow your preferred style.